### ISSUE TITLE
Bug: ABACUS cannot get SCF convergence with combined use of ultrasoft and norm-conserving pseudopotential

### ISSUE THREAD

**Issue by kirk0830** [deepmodeling/abacus-develop#5596]
### Describe the bug

```
                                                                                     
                              ABACUS v3.8.1

               Atomic-orbital Based Ab-initio Computation at UStc                    

                     Website: http://abacus.ustc.edu.cn/                             
               Documentation: https://abacus.deepmodeling.com/                       
                  Repository: https://github.com/abacusmodeling/abacus-develop       
                              https://github.com/deepmodeling/abacus-develop         
                      Commit: 146a50931 (Mon Oct 21 16:41:35 2024 +0800)

 Mon Nov 25 11:00:06 2024
 MAKE THE DIR         : OUT.ABACUS/
 RUNNING WITH DEVICE  : CPU / Intel(R) Xeon(R) Platinum 8255C CPU @ 2.50GHz
 UNIFORM GRID DIM        : 48 * 48 * 48
 UNIFORM GRID DIM(BIG)   : 48 * 48 * 48
 DONE(0.150851   SEC) : SETUP UNITCELL
 DONE(0.187362   SEC) : SYMMETRY
 DONE(5.36323    SEC) : INIT K-POINTS
 ---------------------------------------------------------
 Ion relaxation calculations
 ---------------------------------------------------------
 SPIN    KPOINTS         PROCESSORS  
 1       1140            16          
 ---------------------------------------------------------
 Use plane wave basis
 ---------------------------------------------------------
 ELEMENT NATOM       XC          
 B       1           
 O       1           
 ---------------------------------------------------------
 Initial plane wave basis and FFT box
 ---------------------------------------------------------
 DONE(5.39719    SEC) : INIT PLANEWAVE
 DONE(5.4094     SEC) : LOCAL POTENTIAL
 DONE(5.76873    SEC) : NON-LOCAL POTENTIAL
 MEMORY FOR PSI (MB)  : 82.9742
 DONE(5.82108    SEC) : INIT BASIS
 -------------------------------------------
 STEP OF RELAXATION : 1
 -------------------------------------------
 START CHARGE      : atomic
 DONE(5.87049    SEC) : INIT SCF
 * * * * * *
 << Start SCF iteration.
 ITER       ETOT/eV          EDIFF/eV         DRHO     TIME/s
 DA1     -5.15988845e+02   0.00000000e+00   1.4746e-01  27.96
 DA2     -1.20301889e+04  -1.15142001e+04   5.8264e+01  30.88
 DA3     -2.46284030e+05  -2.34253841e+05   6.4471e+01  43.78
 DA4     -4.00052587e+06  -3.75424184e+06   6.6798e+01  35.23
 DA5     -6.40785857e+07  -6.00780598e+07   1.4915e+01  31.69
 DA6     -1.02532576e+09  -9.61247171e+08   2.4406e+01  23.65
 DA7     -1.64052839e+10  -1.53799582e+10   2.3530e+00  28.55
 DA8     -2.62484615e+11  -2.46079331e+11   5.4242e-02  21.42
 DA9     -4.19975391e+12  -3.93726930e+12   1.1759e-03  22.42

 notconv = 12
 DiagoDavid::diag', too many bands are not converged! 

 notconv = 11
 DiagoDavid::diag', too many bands are not converged! 

 notconv = 12
 DiagoDavid::diag', too many bands are not converged! 

 notconv = 14
 DiagoDavid::diag', too many bands are not converged! 

 notconv = 11
 DiagoDavid::diag', too many bands are not converged! 

 notconv = 15
 DiagoDavid::diag', too many bands are not converged! 

 notconv = 14
 DiagoDavid::diag', too many bands are not converged! 

 notconv = 11
 DiagoDavid::diag', too many bands are not converged! 
```

### Expected behavior

Ultrasoft pseudopotential can be used together with norm-conserving, this is at least supported by QE.

### To Reproduce

[703c4655-fddc-4a9d-a47b-b499a1414d1c.zip](https://github.com/user-attachments/files/17896793/703c4655-fddc-4a9d-a47b-b499a1414d1c.zip)

** Now ALL my calculations show this behavior, the above is just the one I randomly pick **

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [ ] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

**Comment by kirk0830** [comment]
this blocks the test under framework of ABACUS Pseudopot-Nao Square project.

**Comment by YuLiu98** [comment]
I tested with different versions and found some points:
1. there are some modifications between `v3.7.5` and `v3.8.0`, so that cg and dav methods iterate charge density differently.
2. The cg methods can converge successfully, indicating that the combination between ncpp and uspp works.

[liuyu_test.tar.gz](https://github.com/user-attachments/files/17899052/liuyu_test.tar.gz)

**Comment by YuLiu98** [comment]
I have fixed it in PR #5765 
This is my tests example: [issue5596_20241225.tar.gz](https://github.com/user-attachments/files/18247984/issue5596_20241225.tar.gz)
